source/common/stats/stats_impl.h

```diff
@@ -366,7 +395,22 @@ class HistogramImpl : public Histogram, public MetricImpl {
   // Stats::Histogram
   void recordValue(uint64_t value) override { parent_.deliverHistogramToSinks(*this, value); }
 
+  // TODO(ramaraochavali): split the Histogram interface in to two - parent and tls.
+  void merge() override { NOT_IMPLEMENTED; }
+
+  bool used() const override { return true; }
+
+  const HistogramStatistics& intervalStatistics() const override { return interval_statistics_; }
+
+  const HistogramStatistics& cumulativeStatistics() const override {
+    return cumulative_statistics_;
+  }
+
   Store& parent_;
```

Comment:
I realize this is pre-existing, but can you add a TODO to make this private and doc what the storage ownership is?